pasqal_cloud/client.py

```diff
@@ -108,17 +109,28 @@ def _request(
         payload: Optional[Union[Mapping, Sequence[Mapping]]] = None,
```

Comment:
Why not use a dedicated library for retry logic? 
Something like [tenacity](https://tenacity.readthedocs.io/en/latest/)

Reply:
Why add another 3rd party dependency when the logic is rather simple

Reply:
Okay, if you prefer implementing the feature, let's do it. 

But maybe we could draw inspiration from them and separate the retry logic from the method we want to retry? Or do you think it's not worth it since we're only doing it in one place?